Sources/MapboxCoreNavigation/CongestionLevel.swift

```diff
@@ -97,4 +105,34 @@ extension RouteLeg {
 
         return congestionLevels
     }
+    
+    /**
+     Return an array containing the traffic congestion level along each road segment in the route leg geometry.
+
+     The array is formed either by converting values of `segmentNumericCongestionLevels` to `CongestionLevel` type by congestionMapping,
+     or by taking `segmentCongestionLevels`, depening whether `AttributeOptions.numericCongestionLevel` or `AttributeOptions.congestionLevel`
+     was specified in `DirectionsOptions.attributes` during Directions request.
+
+     - parameter congestionMapping: The `CongestionMapping` closure to map `NumericCongestionLevel` and `MapboxStreetsRoadClass` to `CongestionLevel`.
+     - returns: An array containing the traffic congestion level along each road segment in the route leg geometry.
+     */
+    public func resolvedCongestionLevels(congestionMapping: CongestionMapping? = nil) -> [CongestionLevel]? {
```

Comment:
We may allow customizing `CongestionRange`s per `RoadClass` and make the mapping on our side.